pkg/config/analysis/analyzers/gateway/secret.go

```diff
@@ -79,20 +84,64 @@ func (a *SecretAnalyzer) Analyze(ctx analysis.Context) {
 				continue
 			}
 
-			if !ctx.Exists(gvk.Secret, resource.NewShortOrFullName(gwNs, cn)) {
+			secret := ctx.Find(gvk.Secret, resource.NewShortOrFullName(gwNs, cn))
+			if secret == nil {
 				m := msg.NewReferencedResourceNotFound(r, "credentialName", cn)
 
 				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.CredentialName, i)); ok {
 					m.Line = line
 				}
 
 				ctx.Report(gvk.Gateway, m)
+				continue
+			}
+			if !isValidSecret(secret) {
+				m := msg.NewInvalidGatewayCredential(r, r.Metadata.FullName.Name.String(), gwNs.String())
+
+				if line, ok := util.ErrorLine(r, fmt.Sprintf(util.CredentialName, i)); ok {
+					m.Line = line
+				}
+
+				ctx.Report(gvk.Secret, m)
 			}
 		}
 		return true
 	})
 }
 
+func isValidSecret(secret *resource.Instance) bool {
+	s, ok := secret.Message.(*corev1.Secret)
+	if !ok {
+		return false
+	}
+	certData, ok := s.Data[ca.TLSSecretCACertFile]
```

Comment:
This is not accurate, we have more types of valid secrets.

See extractKeyCertAndStaple

Reply:
Thanks, revised. I also noticed that there was already a verification logic, and I improved and reused it to ensure consistency.

Reply:
@howardjohn PTAL, thanks